### ISSUE TITLE
Downloading MNIST fails due to HTTP Error 503

### ISSUE THREAD

**Issue by akihironitta** [PyTorchLightning/lightning-bolts#632]
## 🐛 Bug
http://yann.lecun.com/exdb/mnist/index.html often gets unavailable, and because of the issue, some of the tests fail in Bolts due to 503. This is a known issue https://github.com/pytorch/vision/issues/3549 and its patch will be included in 0.9.1.

### To Reproduce
https://dev.azure.com/PytorchLightning/lightning%20Bolts/_build/results?buildId=7496

**Comment by akihironitta** [comment]
As far as I investigated, there are two known issues about MNIST downloading:
- 403 (https://github.com/pytorch/vision/issues/1938)
- 503 (https://github.com/pytorch/vision/issues/3549)

For 403 error, we can fix it by setting the user agent to avoid the cloudflare filter as done in PL: https://github.com/PyTorchLightning/pytorch-lightning/pull/6357

For 503 error, we need to use a different URL than http://yann.lecun.com/exdb/mnist/ because the website is very unstable and its patch is included in `torchvision>=0.9.1`.

**Comment by oke-aditya** [comment]
The patch is already in stable torchvision, it redirects to a new mirror.

**Comment by akihironitta** [comment]
@oke-aditya Yes, I'm aware of the mirror link. However, I don't think fixing the version of `torchvision` at 0.9.1 only for this issue would be a reasonable fix as it requires users to update their `torchvision`.

**Comment by oke-aditya** [comment]
Yep I don't think so that's the way to use. What possibly can be done is to copy paste `dataset.mnist` from torchvision, and just use it.
